osconfig_tests/test_suites/patch/patch.go

```diff
@@ -0,0 +1,186 @@
+			assertTimeout: 1200 * time.Second,
+		},
+	}
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range testSetup {
+		wg.Add(1)
+		go patchTestCase(ctx, setup, tests, &wg, logger, testCaseRegex, testProjectConfig)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runCreatePatchTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *patchTestSetup, logger *log.Logger, testProjectConfig *testconfig.Project) {
+
+	client, err := daisyCompute.NewClient(ctx)
+	if err != nil {
+		testCase.WriteFailure("error creating client: %v", err)
+		return
+	}
+
+	testCase.Logf("Creating instance with image %q", testSetup.image)
+	var metadataItems []*api.MetadataItems
+	metadataItems = append(metadataItems, testSetup.startup)
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("os-patch-enabled", "true"))
+	inst, err := utils.CreateTestInstance(metadataItems, client, "n1-standard-4", testSetup.image, testSetup.name, testProjectConfig.TestProjectID, testProjectConfig.TestZone, testProjectConfig.ServiceAccountEmail, testProjectConfig.ServiceAccountScopes)
+	if err != nil {
+		testCase.WriteFailure("Error creating instance: %v", utils.GetStatusFromError(err))
+		return
+	}
+	defer inst.Cleanup()
+
+	testCase.Logf("Waiting for agent install to complete")
+	if err := inst.WaitForSerialOutput("osconfig install done", 1, 5*time.Second, 5*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for osconfig agent install: %v", err)
+		return
+	}
+
+	testCase.Logf("Agent installed successfully")
+
+	// create patch job
+	parent := fmt.Sprintf("projects/%s", testProjectConfig.TestProjectID)
+	osconfigClient, err := osconfigserver.GetOsConfigClient(ctx)
+
+	assertTimeout := testSetup.assertTimeout
+
+	req := &osconfigpb.ExecutePatchJobRequest{
+		Parent:      parent,
+		Description: "testing default patch job run",
+		Filter:      fmt.Sprintf("name=\"%s\"", testSetup.name),
+		Duration:    &duration.Duration{Seconds: int64(assertTimeout / time.Second)},
+	}
+	job, err := osconfigClient.ExecutePatchJob(ctx, req)
+
+	if err != nil {
+		testCase.WriteFailure("error while executing patch job: \n%s\n", utils.GetStatusFromError(err))
+		return
+	}
+
+	logger.Printf("%v\n", job)
+
+	// assertion
+	tick := time.Tick(5 * time.Second)
+	timedout := time.Tick(testSetup.assertTimeout)
+	for {
+		select {
+		case <-timedout:
+			testCase.WriteFailure("Patching timed out")
+			return
+		case <-tick:
+			req := &osconfigpb.GetPatchJobRequest{
+				Name: job.Name,
+			}
+			res, err := osconfigClient.GetPatchJob(ctx, req)
+			if err != nil {
+				testCase.WriteFailure("error while fetching patch job: \n%s\n", utils.GetStatusFromError(err))
+				return
+			}
+			logger.Printf("%v\n", res)
+			if res.State > osconfigpb.PatchJob_PATCHING {
```

Comment:
This should probably wait until the state is "SUCCEEDED" and fail if it hits any other terminal states.

Reply:
discussed offline. changed to test instance level state